crates/ty_python_semantic/src/types.rs

```diff
@@ -3525,7 +3555,8 @@ impl<'db> Type<'db> {
             | Type::BoundSuper(_)
             | Type::KnownInstance(_)
             | Type::SpecialForm(_)
-            | Type::AlwaysTruthy => Truthiness::AlwaysTrue,
+            | Type::AlwaysTruthy
+            | Type::TypedDict(_) => Truthiness::AlwaysTrue,
```

Comment:
this isn't correct: if the `TypedDict` is empty, all its inhabitants will be falsy:

```py
from typing import TypedDict

class Empty(TypedDict): ...

x: Empty = {}
print(bool(x))  # False
```

Reply:
Oh, thank you!

Reply:
`{}` is also an inhabitant of `TypedDict`s with `total=False` or a `TypedDict` where all items are `NotRequired`. I guess it's best to just return `Ambiguous`/`bool` for now.